### DISCUSSION TITLE
StepLifecycleListener documentation

### DISCUSSION THREAD

**Discussion by vvyushmanov** [allure-framework/allure2#2426] [Documentation]
Hello! Thank you for your great work :)

I haven't found any documentation (neither in the code itself, nor in the official docs) for the [StepLifecycleListener](https://github.com/allure-framework/allure-java/blob/0dcd87b78edd4c31cf69840d27b0d68b995ad3b7/allure-java-commons/src/main/java/io/qameta/allure/listener/StepLifecycleListener.java#L25), although this interface was very useful  when I needed to add attachments to a failed step in my custom framework.

The only current source of how to use it is either [this question](https://github.com/orgs/allure-framework/discussions/1741) or [this post on StackOverflow](https://stackoverflow.com/questions/45624859/allure2-listener-to-output-the-steps-in-console)

Is it possible to update the documentation to briefly describe how to use this interface? Thank you!

**Comment by baev** [comment]
It is already planned, thanks

**Reply by vvyushmanov** [reply]
Great!

Is there any issue/task, where I can track it?
I want to update our own corporate documentation once this is in place.

Thanks!

**Reply by baev** [reply]
Documentation not open-sourced (yet?). You can track this discussion; I'll post an update here when it's done

**Reply by baev** [reply]
The reason LifecycleListeners are not documented yet is more logical than technical—we need to decide where and in what form to include them in the current documentation structure. 

I see it as some kind of `Guides` or `How to` section that will describe advanced features for different languages and frameworks.

**Reply by vvyushmanov** [reply]
Alright, got it! Thx.
I think a How-to section with language sub-sections would be perfect.
